avalon/nuke/lib.py

```diff
@@ -0,0 +1,230 @@
+import os
+import contextlib
+import nuke
+import re
+import logging
+import toml
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def maintained_selection():
+    previous_selection = nuke.selectedNodes()
+    try:
+        yield
+    finally:
+        # unselect all selection in case there is some
+        current_seletion = nuke.selectedNodes()
+        [n['selected'].setValue(False) for n in current_seletion]
+        # and select all previously selected nodes
+        if previous_selection:
+            [n['selected'].setValue(True) for n in previous_selection]
+
+
+def reset_selection():
+    nodes = nuke.allNodes()
```

Comment:
If this command is required... note that it could be as simple as:
```python
[n['selected'].setValue(False) for n in nuke.selectedNodes()]
```

_Personal note:  Also, somehow I'm more inclined to the name `clear_selection` instead of `reset_selection`, but I guess that's a random preference. :)_

Reply:
I've actually removed both reset selection and select nodes as they weren't used anywhere and are not needed. Must have stayed behind from testing.